streams/src/main/java/org/apache/kafka/streams/processor/TopologyBuilder.java

```diff
@@ -536,6 +537,19 @@ public synchronized final TopologyBuilder connectProcessorAndStateStores(String
         return this;
     }
 
+    public synchronized final TopologyBuilder connectSourceStoreAndTopic(String sourceStoreName, String topic) {
```

Comment:
I'd suggest we move it as part of the `public <K, V> KTable<K, V> table` call or just make it protected, which can still be accessed by the extended `KStreamBuilder` class. The main reason is that as a public function, its semantics will be very confusing to users who observe this function in java doc but do not really know how to use it.